index.js

```diff
@@ -1 +1,207 @@
-console.log('hello world');
+import methodOverride from 'method-override';
+import express from 'express';
+import {
+  read, add, deleteContent, edit,
+} from './jsonFileStorage.js';
+
+const app = express();
+const PORT = 3000;
+const FILENAME = 'data.json';
+
+// Set view engine to EJS
+app.set('view engine', 'ejs');
+// Middleware to allow static images/css files to be served
+app.use(express.static('public'));
+// Provides encoding to url queries etc.
+app.use(express.urlencoded({ extended: false }));
+// Middleware that allows POST methods to be overriden to perform PUT and DELETE requests
+app.use(methodOverride('_method'));
+
+// Helper Function that converts request.body to an Iterable Obj
+const convertToIterableObj = (nonIterableObj) => {
+  const newObj = {};
+  // Convert request.body into an iterable object
+  Object.entries(nonIterableObj).forEach(([key, value]) => {
+    newObj[key] = value;
+  });
+  return newObj;
+};
+
+// Helper Function that checks the inputs for validation
+const performValidationOnRequestBody = (requestBodyObj, callback) => {
+  const inputValidationFeedbackObj = {};
+  let isFormValid = true;
+  const {
+    _, description, date_time, city, state, shape, duration, summary, __,
+  } = requestBodyObj;
+
+  if (description.length < 5) {
+    inputValidationFeedbackObj.description = 'Description is too short!';
+    isFormValid = false;
+  } else {
+    inputValidationFeedbackObj.description = 'ok';
```

Comment:
Ah i see that the "ok" value is used to detect which fields were invalid in the EJS template. Perhaps a more robust solution would be to have the values in `InputValidationFeedbackObj` be objects, where each object contains 2 keys, something like `isValid` and `invalidFieldDescription`. Then the EJS template can check the `isValid` boolean for each field instead of checking whether the field's value is "ok", which seems a bit brittle.